### ISSUE TITLE
Option to skip default values in `JsonSchema` encoders 

### ISSUE THREAD

**Issue by harpocrates** [endpoints4s/endpoints4s#887]
The current handling of default values in `JsonSchema` encoders means that all the fields that have default values will have those values populated in the encoder output. It would be nice to have some options such that the generated encoder would instead skip encoding a field if its value is the default.

This has two potential benefits:

  1. OpenAPI examples can omit optional fields
  2. Generated clients can send smaller payloads
  
The downside is that you end up needing to do an equality check on the value of the field (and if that field is a big nested class, that could be expensive). My main motivation is for 1 though, in which case the extra equality checks occur only once when docs are generated.
